### ISSUE TITLE
possible to add 2 people with the same name different e-mail

### ISSUE THREAD

**Issue by pombase-admin** [pombase/canto#593]
We have a few people in the tool twice

A warning should appear if we try to add a name which is already present.

At the moment we don't have any poeple with identical names so we can figure out a way to disambiguate later. Any of the existing people with the same name will need their "accounts" merged

Original comment by: ValWood

**Comment by ValWood** [comment]
putting at "future" because it happens rarely enough to fix manually for a while

**Comment by ValWood** [comment]
I think Minor Yoshida is now in the tool twice

**Comment by ValWood** [comment]
silke Hauf is in twice 
vt.edu is the correct one

**Comment by ValWood** [comment]
transferring from
https://github.com/pombase/canto/issues/983
All these email address appear in Annotation data.curation field in the database:

a.lock@ucl.ac.uk
A.lock@ucl.ac.uk
Antonia.nilsson@gmail.com

val@sanger.ac.uk
vw253@cam.ac.uk
Fix them to be consistent.

I managed to get a couple of duplicated accounts, both with sessions attached

Hugh Cam hugh.cam@bc.edu 3 
Hugh P. Cam camh@bc.edu 1

Julie Cooper Julie julie.cooper@cancer.org.uk user 5
Julie Cooper Julie julie.cooper@nih.gov user 2
Julie’s current address is julie.cooper@nih.gov

What is the easiest way to merge them?

from duplicate #593

We have a few people in the tool twice

Damien Hermand
and another one I spotted, but I have forgotten who it was...

I think a warning should appear if we try to add a name which is already present.
At the moment we don't have any poeple with identical names so we can figure out a way to disambiguate later. Any of the existing people with the same name will need their "accounts" merged

**Comment by ValWood** [comment]
1. Open a new ticket with what needs to be done

2. Check all the people mentioned and make sure they only have a single account

Damien Hermand
Hugh Cam hugh.cam@bc.edu 3  Hugh P. Cam camh@bc.edu 1
Julie Cooper Julie julie.cooper@cancer.org.uk user 5 Julie Cooper Julie julie.cooper@nih.gov user 2 (current)
a.lock@ucl.ac.uk A.lock@ucl.ac.uk Antonia.nilsson@gmail.com
val@sanger.ac.uk vw253@cam.ac.uk
silke Hauf is in twice vt.edu is the correct one
Minoro Yoshida
Isabelle Jordain  isabelle.jordainbiotek.uio.no imm.jordainbiotek.uio.no
Hirofumi Aila  aibaagr.nagoya-u.ac.jp aibaps.nagoya-u.ac.jp
Dan Mulvihill d.p.mulvihillkent.ac.uk D.P.Mulvihillkent.ac.uk

**Comment by ValWood** [comment]
It doesn't matter if this doesn't get done soon,

biut it would be useful if the accounts could be combined

For eg, I have 4 sessions assigned to Julie Cooper attached to  julie.cooper@cancer.org.uk instead of   julie.cooper@nih.gov	. I can't edit myself because of the existing account.....I get an error...

**Comment by ValWood** [comment]
I think this should happen quite soon....
We are spawning new users as the primary curator often reassigns, and can easily use  slght username and e-mail variations.....
@kimrutherford let us know if you would like a new ticket  
Once the changes are done I'll scan through for the existing obvious duplicates to fix

**Comment by ValWood** [comment]
Kim was asking what people think about autocomplete on names? Would this be OK?

**Comment by ValWood** [comment]
So here we think we might need to do 2 different things.

1. For self selection
First ask
New user?
Existing user?

Then for  Returning user, 
only needs to type e-mail (will pop up their name)
If not recognised try another e-mail, or if your primary e-mail has changed, contact the help desk

New user?
They will need to supply Name and e-mail
If they already exist tell them….

2.
At the point where a PI reassigns a paper, the text will need to be more like Midori's suggestion above, but tweaked so it isn't in the first person.
(can we use autocomplete here? this will be quite hidden, and it will help  stop the situations where the PI uses a variation of both the e-mail and the name (which seems to be the main source of duplications).

**Comment by ValWood** [comment]
Kim also needs to look what happens to existing sessions when I update a user record with a new e-mail. Do they get reassigned to use the new (e-mail) (they should)

**Comment by ValWood** [comment]
- [ ] Once kim is done, Val check for existing diuplicates
- [ ] Kim  merge existing suplicates with data
- [] Kim check that when we update a user e-mail sessions get update

**Comment by kimrutherford** [comment]
I'd love to fix this by requiring ORCIDs.  If we knew the ORCID we wouldn't need to query the user for their name, we could get it from orcid.org.  I think we'd still need to ask for the email.

But I guess insisting on ORCIDs for all curators is a bit harsh and might put people off?

**Comment by ValWood** [comment]
That would be nice, but I'm not sure that we can.

I was hoping that at the point  when people sign up for ORCID, we would also explain that they are signing up to allow us to share the data (i.e. display curator). But I guess we can use ORCID for all of those who have signed up.

So, for example, we could allow biomedcentral, ORCID, etc to link to curation sessions (or a version of), before we have complete literature pages.

Midori, any thoughts? Do you think it would be possible to insist?

**Comment by kimrutherford** [comment]
I've added a script to merge duplicate accounts.  That should make it less error prone.

**Comment by ValWood** [comment]
I sent  Kim a list of the known duplicates with the preferred address.

**Comment by ValWood** [comment]
I don't know if there is anything to discuss here. However, it isn't high priority. We can manage this by spotting duplicate users when they arise  in the short term... it isn't a big deal for me to  keep track of this  for now...

**Comment by ValWood** [comment]
Might be worth looking into this becasue we do quite often get duplicate users.
